app/controllers/sessions_controller.rb

```diff
@@ -4,7 +4,9 @@ def new
 
   def create
     @user = User.find_by(email: params[:session][:email].downcase)
-    if @user && @user.authenticate(params[:session][:password])
+    if @user &&
+       @user.authenticate(params[:session][:password]) &&
+       @user.email_confirmed
```

Comment:
This is pretty ugly. It's also in conflict with my earlier refactoring for interactors. It seemed like the best solution was to add the additional `email_confirmed` call here, knowing that it will produce a conflict and need to be moved into the interactor. Alternatively I could have opened a new PR branched off of this one but that seemed unnecessary.